.github/workflows/main.yml

```diff
@@ -3,13 +3,46 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches:
+      - main
     tags:
       - '*.*.*'
 
 jobs:
+  test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
just wondering why you are testing several python versions 🤔

Reply:
because I could, I guess.  ;) 

If I did not specify python version at all, I could not be sure whether I tested the code on the python version used in the docker image. If I specified only "3.11", I would find it to be unnecessarily restrictive and a little bit ignorant towards the people who might be interested trying to run the code locally without too much additional hassle with setting up the exact environment I want them to have,... (I know that as long as I use a specific version of pandas I do not relax the requirements anywhere close to anything useful from this point of view that's why "because I could" is still the best answer that I can give ;))

Reply:
Closing for now, we can revisit this later on.